### PR TITLE
ENT-1257

### PR DESCRIPTION
**Pull request by muhammad-ammar**
[ENT-1257](https://openedx.atlassian.net/browse/ENT-1257)

These changes will add support to create data reports in `json` format for only `progress_v2` data type. `progress` data type is deprecated so we don't need changes for it.

### CODE REVIEW

**Comment by muhammad-ammar** [comment]
@georgebabey @zubair-arbi  This is ready for review.

File: enterprise_reporting/reporter.py [hunk 1]
```diff
@@ -197,6 +197,18 @@ def _generate_enterprise_report_catalog_json(self):
             json.dump(list(content_metadata), data_report_file, indent=4)
         return [data_report_file]
 
+    def _generate_enterprise_report_progress_v2_json(self):
```
Comment (zubair-arbi) [inline enterprise_reporting/reporter.py]: Just a small suggestion to move this method up just below `_generate_enterprise_report_progress_v2_csv`.

**Review by zubair-arbi — APPROVED** [review]
Changes looks good. After testing 👍

File: enterprise_reporting/reporter.py (hunk 1 above)
Comment (muhammad-ammar) [reply enterprise_reporting/reporter.py]: moved the method

**Review by georgebabey — APPROVED** [review]

File: enterprise_reporting/reporter.py [hunk 2]
```diff
@@ -158,6 +158,18 @@ def _generate_enterprise_report_progress_v2_csv(self):
                 writer.writerow(list(OrderedDict(sorted(enrollment.items())).values()))
         return [data_report_file]
 
+    def _generate_enterprise_report_progress_v2_json(self):
```
Comment (mushtaqak) [inline enterprise_reporting/reporter.py:161]: Did we need version bump on `enterprise_reporting/__init__.py` as well?

**Pull request merged**